### PR TITLE
multi version support

### PR DESCRIPTION
**Pull request by rgolangh**
Motivation
Have a per-version documentation.

Modification
With this change the content/docs directory structure is changed
to content/$version/docs plus few other changes to the front-matter and
hugo.yaml to include the necessary changes to render a Documentation
drop-down menu.

The 'Blog' section remains the same, as blogs are have their own pace
and are not meant to be fixed per version, blogs are articles, not
documentation.

Action items for the team: version 1.1.x now has a folder with a copy of
the main documentation and it needs to be updated accordingly. Also some
of the changes in 'main' may be 1.1.x specific and may need a change
just as well.
notice the change in the directory structure - `docs` is now under a folder and the declaration `hugo.yaml` renders the dropdown and links accordingly 

```yaml
  versions:
    - version: 1.2-rc
      url: '/1.2-rc/docs'
    - version: v1.1.x
      url: '/v1.1.x/docs'
```
content/
├── 1.2-rc
│   ├── docs
│   └── _index.md
├── blog
│   ├── developer-experience-detailed.md
│   ├── developer-experience.md
│   ├── devmode-vs-prodmode.md
│   ├── images
│   ├── _index.md
│   ├── installation-on-existing-rhdh.md
│   └── sonata-operator.md
├── _index.md
└── v1.1.x
    ├── docs
    └── _index.md

7 directories, 9 files

### CODE REVIEW

**Comment by rgolangh** [comment]
image

image

image

**Comment by rgolangh** [comment]
@gciavarrini @masayag This is functioning as easily as before and looks like gets the job done for multi version for documentation.

File: hugo.yaml [hunk 1]
```diff
@@ -38,8 +41,17 @@ params:
       [CC BY 4.0](https://creativecommons.org/licenses/by/4.0) |
     from_year: 2023
       #privacy_policy: link to privacy policy
-  version_menu: Releases
+
+  version_menu: Documentation 
+  version_menu_pagelinks: false
   url_latest_version: https://parodos.dev
+  versions:
+    - version: main
```
Comment (masayag) [inline hugo.yaml:49]: why `main` and not `1.2-rc` ?

File: hugo.yaml [hunk 2]
```diff
@@ -38,8 +41,17 @@ params:
...
+      url: '/main/docs'
+    - version: v1.1.x
+      url: '/v1.1.x/docs'
+  version: main
```
Comment (masayag) [inline hugo.yaml:53]: For now,  if this is the default selected version, I'd go with `v1.1.x` which is the latest stable version available.

**Review by masayag — COMMENTED** [review]
Well done! few comments.

**Comment by gciavarrini** [comment]
Good job!

File: hugo.yaml (hunk 1 above)
Comment (gabriel-farache) [reply hugo.yaml:49]: I think `main` reflects the current work in progress, maybe `SNAPSHOT` would be better? Or do we want to stick as much as possible with the branch naming we have in the repo?

**Comment by gabriel-farache** [comment]
Nice
Maybe it would be nice to have a `stable` and `snapshot` versions too

**Comment by rgolangh** [comment]
Can I merge?

**Review by masayag — APPROVED** [review]

**Pull request merged**